flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/subscriber/KafkaSubscriber.java

```diff
@@ -38,9 +38,11 @@
  * </ol>
  *
  * <p>The KafkaSubscriber provides a unified interface for the Kafka source to support all these
- * three types of subscribing mode.
+ * three types of subscribing mode. {@link
+ * org.apache.flink.connector.kafka.source.enumerator.KafkaSourceEnumerator} only supports adding
+ * new splits and not removing splits in split discovery.
  */
-@Internal
+@PublicEvolving
```

Comment:
@PatrickRen Was it indeed intended to make this interface public?

Reply:
This was original email thread from last year: https://www.mail-archive.com/user@flink.apache.org/msg44340.html

Reply:
IMHO it's reasonable to make `KafkaSubscriber` public. As we are exposing new APIs to users, what about starting a discussion in dev and user mailing list?

Reply:
I agree making it public does not seem like a bad idea.